src/Microsoft.Identity.Web.TokenCache/Distributed/MsalDistributedTokenCacheAdapter.cs

```diff
@@ -185,6 +188,11 @@ protected override async Task RemoveKeyAsync(string cacheKey, CacheSerializerHin
                         Logger.BackPropagateL2toL1(_logger, memoryCacheEntryOptions.Size ?? 0);
                         _memoryCache.Set(cacheKey, result, memoryCacheEntryOptions);
                         Logger.MemoryCacheCount(_logger, _memoryCacheType, read, _memoryCache.Count);
+
+                        if (telemetryData != null)
```

Comment:
nit: `telemetryData?.CacheLevel = ...`

Reply:
using `telemetryData?` on the left side of the `=` does not work for me.